category/functor.lean

```diff
@@ -0,0 +1,106 @@
+/-
+Copyright (c) 2017 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Simon Hudon
+
+Standard identity and composition functors
+-/
+
+import category.basic
+
+universe variables u v w u' v' w'
+
+section functor
+
+variables {F : Type u → Type v}
+variables {α β γ : Type u}
+variables [functor F]
+
+lemma functor.id_map' : has_map.map id = (id : F α → F α) :=
+by { apply funext, apply functor.id_map }
+
+lemma functor.map_comp' (f : α → β) (g : β → γ) :
+  has_map.map (g ∘ f) = (has_map.map g ∘ has_map.map f : F α → F γ) :=
+by { apply funext, intro, apply functor.map_comp }
+
+@[norm]
+lemma functor.map_map (f : α → β) (g : β → γ) (x : F α) :
+  g <$> f <$> x = (g ∘ f) <$> x :=
+by rw ← functor.map_comp
+
+end functor
+
+structure identity (α : Type u) : Type u :=
```

Comment:
As far as I understand this can be just `id` in lean. also `compose` can be just function composition.